### PR TITLE
PYIC-2751 Update the Lambda removing support for API Gateway requests.

### PR DESCRIPTION
**Pull request by rpayal**
## Proposed changes
Update the Lambda removing support for API Gateway requests.
### What changed
Updated (build-proven-user-identity-details) Lambda to remove support for API Gateway requests.

### Issue tracking

- [PYIC-2751](https://govukverify.atlassian.net/browse/PYIC-2751)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-2751]: https://govukverify.atlassian.net/browse/PYIC-2751?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by rpayal** [comment]
Tested running regression pack against dev stack.

**Review by chris-dco — APPROVED** [review]

**Review by chris-dco — CHANGES_REQUESTED** [review]
Not currently building.

**Review by TalatAydinGDS — DISMISSED** [review]
LGTM

**Review by TalatAydinGDS — DISMISSED** [review]

File: lambdas/build-proven-user-identity-details/src/test/java/uk/gov/di/ipv/core/buildprovenuseridentitydetails/BuildProvenUserIdentityDetailsHandlerTest.java [hunk 1]
```diff
@@ -577,12 +551,14 @@ private VcStoreItem createVcStoreItem(
...
-        final var requestType = new TypeReference<Map<String, Object>>() {};
-        final var request = objectMapper.convertValue(event, requestType);
-        final var response = underTest.handleRequest(request, context);
+    private static String getJsonResponse(Map<String, Object> response)
+            throws JsonProcessingException {
+        return objectMapper.writeValueAsString(response);
+    }
 
-        return objectMapper.convertValue(response, APIGatewayProxyResponseEvent.class);
+    private String makeRequest(JourneyRequest event, Context context)
```
Comment (chris-dco) [inline lambdas/build-proven-user-identity-details/src/test/java/uk/gov/di/ipv/core/buildprovenuseridentitydetails/BuildProvenUserIdentityDetailsHandlerTest.java]: As an alternative the following may simplify the code which is currently in use in the CheckExistingIdentityHandlerTest:
```java
private <T extends BaseResponse> T handleRequest(
        JourneyRequest request, Context context, Class<T> classType) throws IOException {
    try (var inputStream = new ByteArrayInputStream(mapper.writeValueAsString(request).getBytes());
            var outputStream = new ByteArrayOutputStream()) {
        userTest.handleRequest(inputStream, outputStream, context);
        return mapper.readValue(outputStream.toString(), classType);
    }
}
```

Which can be called like the following.
```java
var journeyResponse = handleRequest(event, context, JourneyResponse.class); // automatically returns a JourneyResponse
```

or:
```java
var journeyResponse = handleRequest(event, context, JourneyErrorResponse.class); // automatically returns a JourneyErrorResponse
```

File: lambdas/build-client-oauth-response/src/test/java/uk/gov/di/ipv/core/buildclientoauthresponse/BuildClientOauthResponseHandlerTest.java [hunk 2]
```diff
@@ -124,12 +127,12 @@ void shouldReturn200OnSuccessfulOauthRequest_withNullIpvSessionAndClientSessionI
                 .thenReturn(getClientOAuthSessionItem());
 
         JourneyRequest event =
-                new JourneyRequest(
-                        null,
-                        TEST_IP_ADDRESS,
-                        TEST_CLIENT_OAUTH_SESSION_ID,
-                        null,
-                        TEST_FEATURE_SET);
+                JourneyRequest.builder()
```
Comment (chris-dco) [inline lambdas/build-client-oauth-response/src/test/java/uk/gov/di/ipv/core/buildclientoauthresponse/BuildClientOauthResponseHandlerTest.java:129]: So pleased to see the use of the builder pattern here :)

**Review by chris-dco — DISMISSED** [review]
Some minor comments. Otherwise working as expected.

**Review by blakeyp — APPROVED** [review]

File: lambdas/build-proven-user-identity-details/src/test/java/uk/gov/di/ipv/core/buildprovenuseridentitydetails/BuildProvenUserIdentityDetailsHandlerTest.java (hunk 1 above)
Comment (rpayal) [reply lambdas/build-proven-user-identity-details/src/test/java/uk/gov/di/ipv/core/buildprovenuseridentitydetails/BuildProvenUserIdentityDetailsHandlerTest.java]: Done

**Pull request merged**